### ISSUE TITLE
Bug: Sub-file staging stages more than wanted

### ISSUE THREAD

**Issue by jflevesque** [ForkIssues/TrackerWin#1607]
Hi,

I've just updated to 1.76.0.0 and I've been getting issues with staging. For example, I renamed a member in a test class, and I'm only staging some of the changes. When I click on the Stage button from the diff window, it often includes changes from a different stage block.

Ex.
```
#block 1            |
-- original.1       | ++ change.1
#end block 1        |
//...
#block 2            |
-- original.2       | ++ change.2
#end block 2        |
```
With the above example, I only want to stage block 2, so I click the stage button from the diff window. In most cases, the staged changes I would be getting is the following:

Unstaged:
```
#block 1            |
                    | ++ change.1
#end block 1        |
```
Staged:
```
#block 1            |
-- original.1       |
#end block 1        |
//...
#block 2            |
-- original.2       | ++ change.2
#end block 2        |
```

**Comment by DanPristupov** [comment]
Hi. Thank you for reporting that.

Do you use side-by-side mode or the normal one?
Does the problem occur if you select the piece of code with manually and stage it?

**Comment by DanPristupov** [comment]
Can you send a screenshot of code sample to support@fork.dev, please? I don't need the whole source code, the beginning of the lines would be enough.

**Comment by jflevesque** [comment]
I use side by side mode.
I'll create a test file with which I have the behavior. Give me a few minutes.

**Comment by jflevesque** [comment]
Ok, I got a file which triggers the bug. Its a bit more specific than I thought. Here are the repro steps:

1. Stage this file
[ForkStagingBug.txt](https://github.com/fork-dev/TrackerWin/files/9236048/ForkStagingBug.txt)
1. Make a find and replace case insensitive "s_" to ""
1. Save file.
1. In the side by side window, stage only the method name changes using the Stage button
You'll notice that it also included the renaming of the variable from the original, without the changes from the unstaged file.

**Comment by DanPristupov** [comment]
> Here are the repro steps:

Am I doing everything right?
![ezgif com-gif-maker](https://user-images.githubusercontent.com/618115/182212101-14a74935-449e-4851-87e8-455e2a60616b.gif)

**Comment by jflevesque** [comment]
Yes. If you look right before and after your Stage click, line 5 changes on the left, because it is staged at the same time as line 12

**Comment by DanPristupov** [comment]
Right! 👍

Although the 1.76 update may look small, it is actually one of the biggest updates in Fork (about 60 commits). Almost all changes were related to staging in both normal and side-by-side mode. Thanks for pointing out the bug, we'll fix it :).

**Comment by DanPristupov** [comment]
We just published the `1.76.5` update.
